### PR TITLE
Br217

### PR DESCRIPTION
**Pull request by kthare10**
Changes for Issue# https://github.com/RENCI-NRIG/neuca-guest-tools/issues/5

neuca-guest changes w.r.t Orca issue RENCI-NRIG/orca5#217

As part of this Issue following changes have been made:
a) Added support for /etc/hosts to be updated for a group of nodes based on IP information available for them on COMET
b) Added support for generation of public keys for root user and add the keys to /etc/authorized keys for a group of nodes based on the keys and groups defined in COMET

The changes for the above two features are implemented to work both on ExoGeni and Chameleon.

### CODE REVIEW

**Comment by kthare10** [comment]
I have identified an issue with the code uploaded in the review in file instancedata.py in functions updateHostsFromComet and updatePubKeysFromComet. With the current implementation, neuca-guest-tools could result in overwriting/removing user defined changes in /etc/hosts and /root/.ssh/authorized_keys files. I am working to address this issue and will upload the fix for same.

**Comment by kthare10** [comment]
@vjorlikowski - Please confirm if you plan to review the changes in this pull request.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+#### PubKeys Context
+`python3 comet_client.py -o create_family -c https://13.59.255.221:8111 -i ./input1.json`
+
+# Usage on exogeni nodes
+## Fetch neuca-guest-tools code by following commands
+```
+git clone https://github.com/RENCI-NRIG/neuca-guest-tools
+cd neuca-guest-tools/neuca-py/
+git checkout br217
```
Comment (ibaldin) [inline README.md]: So will this be needed after the merge?

Comment (kthare10) [reply README.md]: No, this was a workaround only until merge

**Comment by ibaldin** [comment]
@kthare10 I left a comment about line 64 of the README.md file, otherwise looks good.

File: README.md (hunk 1 above)
Comment (kthare10) [reply README.md]: Updated README to remove checkout statement

Comment (ibaldin) [reply README.md]: Since this is the merge, should we remove that :-)

Comment (kthare10) [reply README.md]: Removed it!

**Pull request merged**